esmvaltool/cmorizers/data/downloaders/datasets/persiann_cdr.py

```diff
@@ -0,0 +1,42 @@
+"""Script to download PERSIANN-CDR."""
+
+from datetime import datetime
+
+from dateutil import relativedelta
+
+from esmvaltool.cmorizers.data.downloaders.wget import WGetDownloader
+
+
+def download_dataset(config, dataset, start_date, end_date, overwrite):
+    """Download dataset.
+
+    Parameters
+    ----------
+    config : dict
+        ESMValTool's user configuration
+    dataset : str
+        Name of the dataset
+    start_date : datetime
+        Start of the interval to download
+    end_date : datetime
+        End of the interval to download
+    overwrite : bool
+        Overwrite already downloaded files
+    """
+    if start_date is None:
+        start_date = datetime(1983, 1, 1)
+    if end_date is None:
+        end_date = datetime(2020, 1, 1)
+    loop_date = start_date
+
+    downloader = WGetDownloader(
+        config=config,
+        dataset=dataset,
+        overwrite=overwrite,
+    )
+    base_path = (
+        "https://www.ncei.noaa.gov/data/precipitation-persiann/access/"
+        "{year}/")
+    while loop_date <= end_date:
```

Comment:
For this dataset as well, there seems to be a problem with the loop. Only files for the year specified in `start_date` (here 1983) are downloaded.

Reply:
I need to remove the index.html file manually for wget to be able to download the next files. It looks like it was looking at this file to get the list of files to download